core/lib/src/response/responder.rs

```diff
@@ -217,34 +219,156 @@ impl<'r> Responder<'r> for String {
     }
 }
 
+fn range_response<'r, B: io::Read + io::Seek + 'r>(range: Range, mut body: B) -> Result<response::Response<'r>, B> {
+    use std::cmp;
+    use http::hyper::header::{ContentRange, ByteRangeSpec, ContentRangeSpec};
+
+    match range {
+        Range::Bytes(ranges) => {
+            if ranges.len() == 1 {
+                let (start, end, total) = match ranges[0] {
+                    ByteRangeSpec::FromTo(mut start, mut end) => {
+                        if end < start {
+                            return Ok(
+                                Response::build()
+                                    .status(Status::RangeNotSatisfiable)
+                                    .header(AcceptRanges(vec![RangeUnit::Bytes]))
+                                    .finalize()
+                            );
+                        }
+                        let size = body.seek(io::SeekFrom::End(0))
+                            .expect("Attempted to retrieve size by seeking, but failed.");
```

Comment:
I'm uncomfortable with this `expect`. I think it's fine for `Vec<u8>` and `[u8]` for example, but I'm not so sure about `File`.

Reply:
How would you suggest to "fix" this?

To me this in an "Internal Server Error", because the state of the seek pointer is unknown at this point, which means we cannot fallback anymore, because this will also have an impact on `Read`.
Rocket would catch this (unlikely) panic anyway, but I guess I could create a 500-response instead and return that.

Reply:
Generally, I prefer to return some kind of `Error` directly rather than panicking. Although looking at it, `Response::set_sized_body` itself panics instead of returning `Result`.

I looked at a few of the platform implementations of `Seek` for `File`. It looks like *nix and Windows only panic when the offset is out of range for an `i64`, or for any platform-specific reasons (e.g. the `File` object is in reality a handle to a pipe or socket, not a file).

My original thinking was that in the case of a "seek to get size" error we could just seek back to the start, ignoring the range request, and/or return a range not satisfiable. But I'm not sure if that's really worth the complexity.

Reply:
If one seek already has failed, it is very likely, that another to the beginning will fail as well.

The current approach is my favorite, but I if you would like to see something else, just tell me.